java/core/src/main/java/com/google/protobuf/Utf8.java

```diff
@@ -1554,5 +1767,93 @@ private static int unsafeIncompleteStateFor(long address, final int byte1, int r
     }
   }
 
+  /**
+   * Adaptation of http://bjoern.hoehrmann.de/utf-8/decoder/dfa/
+   *
+   * Copyright (c) 2008-2009 Bjoern Hoehrmann <bjoern@hoehrmann.de>
```

Comment:
I'll consult our internal team about this, but I think we can't accept this code piece unless the author 'Bjoern Hoehrmann' has signed google CLA and explicitly acknowledge the inclusion of his work in protobuf.